e2etest/bgptests/bgp.go

```diff
@@ -126,6 +126,19 @@ var _ = ginkgo.Describe("BGP", func() {
 
 			svc, _ := testservice.CreateWithBackend(cs, f.Namespace.Name, "external-local-lb", tweak)
 			defer testservice.Delete(cs, svc)
+			desiredLbIPs := svc.Annotations["metallb.universe.tf/loadBalancerIPs"]
+			if desiredLbIPs != "" {
+				ingressPts := svc.Status.LoadBalancer.Ingress
+				framework.ExpectNotEqual(len(ingressPts), 0)
+				assignedIPs := strings.Join(func(ingresses []corev1.LoadBalancerIngress) []string {
```

Comment:
This is a bit hard to follow. I'd just produce the []string out of svc.Status.LoadBalancer.Ingress and use it to feed strings.Join, or even better, do that in a `ValidateDesiredLB(svc)` func under the `service` package .

Reply:
sure, removed nested function now and also factored out this whole logic into `ValidateDesiredLB` function in` validate.go`